components-datastewardship/src/main/java/org/talend/components/datastewardship/runtime/writer/TdsTaskWriter.java

```
@@ -65,27 +81,106 @@ public void write(Object datum) throws IOException {
+        if (master) {
+            //add master record
+            for (Schema.Field f : dataSchema.getFields()) {
+                if (!(f.equals(groupIdField) || f.equals(masterField) 
+                        || f.equals(sourceField) || f.equals(scoreField))) {
+                    Object value = record.get(f.pos());
+                    recordObj.put(f.name(), value);
+                }
+            }
+            taskObj.put("record", recordObj); //$NON-NLS-1$
+        } else {
+            //add source record
+            String source = String.valueOf(record.get(sourceField.pos()));
+            Map<String, Object> sourceRecordObj = new HashMap<String, Object>();
+            for (Schema.Field f : dataSchema.getFields()) {
+                if (!(f.equals(groupIdField) || f.equals(masterField) 
+                        || f.equals(sourceField) || f.equals(scoreField))) {
+                    Object value = record.get(f.pos());
+                    recordObj.put(f.name(), value);
+                }
+                sourceRecordObj.put("source", source); //$NON-NLS-1$
+                sourceRecordObj.put("record", recordObj); //$NON-NLS-1$
+            }
+            sourceRecords.add(sourceRecordObj);
+            taskObj.put("sourceRecords", sourceRecords); //$NON-NLS-1$
         }
-        taskObj.put("type", sink.getCampaignType()); //$NON-NLS-1$
-        taskObj.put("record", recordObj); //$NON-NLS-1$
         taskObjs.add(taskObj);
-
-        String resourceToCreate = "api/v1/campaigns/owned/" + sink.getCampaignName() + "/tasks"; //$NON-NLS-1$ //$NON-NLS-2$
-
-        int statusCode = getConnection().post(resourceToCreate, taskObjs.toJSONString());
-        handleResponse(statusCode, resourceToCreate, record);
+    }                               
+   
+    @Override
+    public Result close() {
+        if (taskObjs.size() > 0) {
+            try {
+                submit();
+            } catch (IOException e) {
+                e.printStackTrace();
```

Exception should be emitted not swallowed.